relayer/chains/cosmos/event_parser.go

```diff
@@ -29,28 +29,45 @@ type ibcMessageInfo interface {
 
 func (ccp *CosmosChainProcessor) ibcMessagesFromBlockEvents(
 	beginBlockEvents, endBlockEvents []abci.Event,
-	height uint64,
+	height uint64, base64Encoded bool,
 ) (res []ibcMessage) {
 	chainID := ccp.chainProvider.ChainId()
-	res = append(res, ibcMessagesFromEvents(ccp.log, beginBlockEvents, chainID, height)...)
-	res = append(res, ibcMessagesFromEvents(ccp.log, endBlockEvents, chainID, height)...)
+	res = append(res, ibcMessagesFromEvents(ccp.log, beginBlockEvents, chainID, height, base64Encoded)...)
+	res = append(res, ibcMessagesFromEvents(ccp.log, endBlockEvents, chainID, height, base64Encoded)...)
 	return res
 }
 
-type packetKey struct {
-	sequence uint64
-	channel  processor.ChannelKey
-}
-
 // ibcMessagesFromTransaction parses all events within a transaction to find IBC messages
 func ibcMessagesFromEvents(
 	log *zap.Logger,
 	events []abci.Event,
 	chainID string,
 	height uint64,
+	base64Encoded bool,
 ) (messages []ibcMessage) {
 	for _, event := range events {
-		evt := sdk.StringifyEvent(event)
+		var evt sdk.StringEvent
+		if base64Encoded {
+			evt = sdk.StringEvent{Type: event.Type}
+			for _, attr := range event.Attributes {
+				key, err := base64.StdEncoding.DecodeString(attr.Key)
+				if err != nil {
+					log.Error("Failed to decode legacy key as base64", zap.String("base64", attr.Key), zap.Error(err))
+					continue
+				}
+				value, err := base64.StdEncoding.DecodeString(attr.Value)
+				if err != nil {
+					log.Error("Failed to decode legacy value as base64", zap.String("base64", attr.Value), zap.Error(err))
+					continue
+				}
+				evt.Attributes = append(evt.Attributes, sdk.Attribute{
+					Key:   string(key),
+					Value: string(value),
+				})
```

Comment:
This could be its own function, which would increase readability and maintainability.

Reply:
agreed, moved